### PR TITLE
Only cache query results not single courses

### PR DESCRIPTION
**Pull request by mjfrey**
Only cache offers for queries not single courses.

### CODE REVIEW

**Comment by mjfrey** [comment]
@mattdrayer

**Comment by mattdrayer** [comment]
why do we want to only cache the offers for the catalog query scenario?  Seems like we would still want to cache and avoid making a backend call in either case?  Maybe the hash key could just be different depending on which mode we're operating in?

**Comment by mattdrayer** [comment]
something like:

```
cache_key = 'voucher_offers_{}'.format(voucher.id)

query = voucher.offers.first().benefit.range.catalog_query
if query:
    cache_key = 'voucher_offers_{}'.format(query)

cache_hash = hashlib.md5(cache_key).hexdigest()
offers = cache.get(cache_hash)

if not offers:  # pragma: no cover
    try:
        offers = self.get_offers(products, request, voucher)
    except (ConnectionError, SlumberBaseException, Timeout):
        logger.error('Could not get course information.')
        return Response(status=status.HTTP_400_BAD_REQUEST)

cache.set(cache_hash, offers, settings.COURSES_API_CACHE_TIMEOUT)
```

**Comment by mjfrey** [comment]
@mattdrayer good idea.

**Comment by mattdrayer** [comment]
Green build -- clear to merge -- 👍

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py [hunk 1]
```diff
@@ -145,6 +145,22 @@ def test_voucher_offers_listing_product_found(self):
...
+        new_range, __ = Range.objects.get_or_create(catalog_query='*:*')
+        new_range.add_product(seat)
+        voucher, __ = prepare_voucher(_range=new_range, benefit_value=10)
+        voucher, __ = get_voucher_and_products_from_code(voucher.code)
+        request = self.prepare_offers_listing_request(voucher.code)
+        response = self.endpointView(request)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertGreater(len(response.data), 0)
+        # More assertions here
```
Comment (marjev) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py:161]: Do we need this comment?

**Comment by marjev** [comment]
This can be closed due to the existence of the #819. I included these changes there to avoid having numerous commits in history regarding this API endpoint.

**Pull request closed without merging**